lib/mutant/env/bootstrap.rb

```diff
@@ -106,6 +106,31 @@ def infect
         @integration = config.integration.new(config).setup
       end
 
+      # Return matched subjects
+      #
+      # @return [Enumerable<Subject>]
+      #
+      # @api private
+      #
+      def matched_subjects
+        Matcher::Compiler.call(self, config.matcher).to_a
```

Comment:
The evaluation to `Array` via `#to_a` is needed to force evaluation of all subjects at this point, why we need to force:

 * Evaluation of matchers to subjects can trigger warnings, and I want all warnings to be printed before analysis starts
* When we lazily match later on, the object space might be "infected" with zillions of rspec classes, making matching very slow.